items-service/src/repository/ItemRepository.js

```diff
@@ -0,0 +1,63 @@
+const Usuario = require('../models/Usuario');
+const Item = require('../models/Item');
+
+const RETORNO_SUCESSO = 1;
+const RETORNO_ERRO = 2;
+const RETORNO_ERRO_INESPERADO = 3;
+
+module.exports = {
+   async addItem(descricao, prioridade, dificuldade) {
+      const item = await Item.create({
+         descricao,
+         prioridade,
+         dificuldade,
+         dataCadastro: Date.now(),
+      });
+      return item;
+   },
+   async findItemByDescription(description) {
+      const item = await Item.find({ descricao: description }, () => {});
+      return item;
+   },
+   async findItemById(id) {
+      const item = await Item.findById(id);
+      return item;
+   },
+   async findItemByUser(nomeUser) {
+      const user = await Usuario.find({ nome: nomeUser });
+      const item = await Item.find({ usuarioDesignado: user.id });
+      return item;
```

Comment:
O find vai retornar um array, se vc fizer `[].id` vai dar um erro de função não definida, nesse caso deve-se usar o `findOne` e validar se o user não está nulo

```suggestion
      const user = await Usuario.findOne({ nome: nomeUser });
      if(!user) return new Array();
      const item = await Item.find({ usuarioDesignado: user.id });
      return item;
```